File: caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    // Checking number of command-line arguments
    if (argc != 2)
    {
        printf("Please enter one command-line argument\n");
        return 1; 
    }
    
    // converting command-line argument into integer k
    int k = atoi(argv[1]);
    
    // prompt the user to enter plaintext 
    printf("plaintext: ");
    string plaintext = get_string();
    printf("ciphertext: ");
   
    if (plaintext != NULL)
    {
        // Loop through each character of the user's input
        for (int i = 0, n = strlen(plaintext); i < n; i++)
        {
            int ai = 0, j = 0;
            
            // Shifting letters by k places for uppercase letters
            
            if (isupper(plaintext[i]) != 0)
            {
                ai = (int)plaintext[i]- (int)65;  // alphabetical index number for the letter
                j = (ai + k) % (int)26 + (int)65;
                printf("%c", (char)j);
            }
            
            // Shifting letters by k places for lowercase letters
            
            else if (islower(plaintext[i]) != 0)
            {
                ai = (int)plaintext[i]- (int)97;  // alphabetical index number for the letter
                j = (ai + k) % (int)26 + (int)97;
                printf("%c", (char)j);
            }
            
            // When the input is not a letter, no encryption is made
            else
            {
                printf("%c", plaintext[i]);
            }
                
        }
        // starting a new line
        printf("\n");
    }
    return 0;
}

File: caesar.py
'''
Pset6 - Caesar (less comfortable)
Encrypt message using Caesar's cipher  
CS50
'''

import cs50
import sys

# ensure proper usage
if len(sys.argv) != 2:
    print("Usage: python caesar.py k")
    exit(1)

# convert the command-line argument into an integer
k = int(sys.argv[1])

# prompt user to input plaintext
print("Plaintext: ", end="")
plaintext = cs50.get_string()


# check to see if input value is null
if plaintext != None: 

    # iterate through each character in the user's input
    for i in range(len(plaintext)):
            
        # encrypt uppercase alphabetical characters
        if plaintext[i].isupper():
            num = ord(plaintext[i]) 
            i = num - 65
            ai = (i + k) % 26
            cipher = ai + 65
            print(chr(cipher), end="")
            
        # encrypt lowercase alphabetical characters
        elif plaintext[i].islower():
            num = ord(plaintext[i]) 
            i = num - 97
            ai = (i + k) % 26
            cipher = ai + 97
            print(chr(cipher), end="")
         
        # skip non-alphabetical characters   
        else: 
            print(plaintext[i], end="")
         
    # add a new line when finishing encrypting the entire user input
    print("")
    exit(0)


